source/js/components/post-code-validation.js

```diff
@@ -0,0 +1,58 @@
+// Importing JSON list of countries and post-code info.
+// This info is also shared with donate/forms.py to check on the backend.
+try {
+  var countriesAndPostCodes = require("./post-codes-list");
+} catch (err) {
+  console.error(
+    "Could not import post-code data. Zip code is required by default."
+  );
+}
+// The container for the post-code input field.
+const postCodeInput = document.querySelector(".post-code-input");
+
+// The container that houses both the post code and city inputs.
+const formAndCityContainer = document.querySelector(".form__group--city-post");
+
+// The select object that allows users to select their country.
+const countrySelector = document.getElementById("id_country");
+
+// If you are on the form page, check if users country uses post code.
+if (countrySelector) {
+  checkForCountryPostCode();
+}
+// Event listener for checking post-code when user selects new country.
+function countryPostCodeValidation() {
+  if (countrySelector && countriesAndPostCodes) {
+    countrySelector.addEventListener("change", (e) => {
+      checkForCountryPostCode();
+    });
+  }
+}
+
+function checkForCountryPostCode() {
+  // The users selected country.
+  const selectedCountrysName =
+    countrySelector.options[countrySelector.selectedIndex].text;
+
+  // Finding the country object in the reference array.
+  const countryObject = countriesAndPostCodes.find(
+    (country) => country.name === selectedCountrysName
+  );
+
+  if (countryObject !== undefined) {
+    if (countryObject.postal) {
+      // Display post code field.
+      formAndCityContainer.classList.remove(
+        "form__group--city-post--full-width"
+      );
```

Comment:
what does this line do, and is there a different way we can achieve the same thing? The fact that it hardcodes a CSS class including the words "full width" feels like it's one template/scss change way from being a bug we need to hunt down =)

Reply:
Hi Pomax! I have removed any of the hardcoding of css classes, and instead  change the css value of the grid in the for the container from "0.3fr 0.7fr" to "auto" to allow the city input to take the full width of the form.